renderer/index.js

```diff
@@ -257,10 +276,27 @@ function isNotTorrentFile (file) {
 function addTorrent (torrentId) {
   if (!torrentId) torrentId = 'magnet:?xt=urn:btih:6a9759bffd5c0af65319979fb7832189f4f3c35d&dn=sintel.mp4'
   var torrent = startTorrenting(torrentId)
-  if (state.saved.torrents.find((x) => x.infoHash === torrent.infoHash)) {
-    return // torrent is already in state.saved
+
+  // check if torrent is duplicate
+  var exists = state.saved.torrents.find((x) => x.infoHash === torrent.infoHash)
+  if (exists) return window.alert('That torrent is already downloading.')
+
+  // save only if infoHash is available
+  if (torrent.infoHash) {
+    state.saved.torrents.push({
+      infoHash: torrent.infoHash
+    })
+  } else {
+    torrent.on('infoHash', () => saveTorrentData(torrent))
```

Comment:
Good catch. Remote torrents, like those that come from filesystem paths or http urls will not have the `infoHash` property set immediately :)